lib/elixir/test/elixir/kernel/errors_test.exs

```diff
@@ -433,6 +433,18 @@ defmodule Kernel.ErrorsTest do
     assert_eval_raise CompileError, "nofile:1: undefined function call/2", 'call foo, do: :foo'
   end
 
+  test "bodiless local function" do
+    ExUnit.CaptureIO.capture_io(:stderr, fn ->
```

Comment:
I know that there is an on going discussion on #9455, but I couldn't find another way to remove the warnings, maybe I should remove the `async: true`?